throughs/flatten.js

```diff
@@ -33,9 +39,9 @@ module.exports = function flatten () {
         read(null, function (end, stream) {
           if(end)
             return cb(end)
-          if(Array.isArray(stream) || stream && 'object' === typeof stream)
+          if(stream && 'object' === typeof stream)
             stream = values(stream)
-          else if('function' != typeof stream)
+          else if ('function' !== typeof stream)
```

Comment:
These lines, 42 and 44, are good to keep. Would be especially good to replace double-equality with triple-equality where ever we find them, they actually have a performance benefit, and don't hurt readability.